SPECS/kube-vip-cloud-provider/kube-vip-cloud-provider.spec

```
@@ -15,21 +15,22 @@ Source0:        https://github.com/kube-vip/%{name}/archive/refs/tags/v%{version
 # Adding the vendor folder and creating a tarball
 # How to re-build this file:
 # 1. wget https://github.com/kube-vip/%%{name}/archive/refs/tags/v%%{version}tar.gz -O %%{name}-%%{version}.tar.gz
-# 2. tar -xf %%{name}-%%{version}.tar.gz
-# 3. cd %%{name}-%%{version}
-# 4. go mod vendor
-# 5. tar -cf %%{name}-%%{version}-vendor.tar.gz vendor
+# 2. <repo-root>/toolkit/scripts/build_go_vendor_cache.sh %%{name}-%%{version}.tar.gz
 
 Source1: %{name}-%{version}-vendor.tar.gz
 
-BuildRequires: golang
+Patch1:        CVE-2023-47108.patch
+
+BuildRequires: golang >= 1.22
 
 %description
 The Kube-Vip cloud provider functions as a general-purpose cloud provider for on-premises bare-metal or virtualized setups. 
 
 %prep
-%setup -q
-tar -xvf %{SOURCE1}
+%autosetup -N
+# Apply vendor before patching
+tar --no-same-owner -xf %{SOURCE1}
+%autopatch -p1
```

Suggestion to avoid manual steps:
```suggestion
%autosetup -a 1 -p1
```

---

Awesome. I looked for something that would do that but couldn't find it.